torchinductor/lowering.py

```diff
@@ -1452,6 +1452,8 @@ def full(size, fill_value, **kwargs):
 def gather(x, dim, index):
     assert isinstance(x, TensorBox)
     assert isinstance(dim, int)
+    if dim < 0:
+        dim = len(x.get_size()) + dim
     assert "int" in str(index.get_dtype())
```

Comment:
`assert index.get_dtype() == torch.int64`, other integer types are not supported in eager